lib/purchases_flutter.dart

```diff
@@ -175,13 +175,13 @@ class Purchases {
   /// [upgradeInfo] Android only. Optional UpgradeInfo you wish to upgrade from
   /// containing the oldSKU and the optional prorationMode.
   static Future<PurchaserInfo> purchasePackage(Package packageToPurchase,
-      {UpgradeInfo upgradeInfo}) async {
+      {UpgradeInfo? upgradeInfo}) async {
     var response = await _channel.invokeMethod('purchasePackage', {
       'packageIdentifier': packageToPurchase.identifier,
       'offeringIdentifier': packageToPurchase.offeringIdentifier,
       'oldSKU': upgradeInfo != null ? upgradeInfo.oldSKU : null,
       'prorationMode': upgradeInfo != null && upgradeInfo.prorationMode != null
-          ? upgradeInfo.prorationMode.index
+          ? upgradeInfo.prorationMode!.index
```

Comment:
same question here, does this need to also be unwrapping `upgradeInfo`?

Reply:
Here we do need unwrapping as `prorationMode` could be implemented like this:

```dart
String? get prorationMode {
  if (random() < 0.5) return null;
  return 'RevenueCat';
}
```

And the analyzer can't be sure that every time `upgradeInfo.prorationMode` is called the value is always null.

What I would suggest to do instead is this:

```dart
final prorationMode = upgradeInfo?.prorationMode;

prorationMode != null ? prorationMode.index : null
```

With that we don't need to assert that it's not null and the analyzer knows it itself.

Reply:
agreed. @vegaro we should do the updates here, even if we're safe for now since we know the current implementation of upgradeInfo won't change values for `prorationMode` in between calls